### PR TITLE
completed issues EB-001, EB-002, EB-003. database generic fields externalized, code cleanup

### LINKED ISSUE #9: EB-001: Include Extra field to login response
Include: 
* companyname,
* recentInvoices

### LINKED ISSUE #10: EB-002: API's should return records in descending order
API for
invoices, estimates, work report, should return in descending order

### LINKED ISSUE #11: EB-003: CRD Operation for Vat type
The Tax code in terms of percentage having percentage, label and tag
eg. 20% Vat 
      10% Vat

Create 
Read
Delete

### PR DESCRIPTION
**Pull request by ChibuezePaul**

### CODE REVIEW

File: src/main/java/uk/co/setech/easybook/model/BaseEntity.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.annotations.UpdateTimestamp;
+
+import java.time.LocalDateTime;
+
+@Data
+@MappedSuperclass
+public abstract class BaseEntity {
+    @Id
+    @GeneratedValue//(strategy = GenerationType.IDENTITY)
```
Comment (ChibuezePaul) [inline src/main/java/uk/co/setech/easybook/model/BaseEntity.java:17]: WE NEED TO SWITCH TO IDENTITY

Comment (amoz28) [reply src/main/java/uk/co/setech/easybook/model/BaseEntity.java:17]: I can see you done it already, why not make it for everything?

File: src/main/java/uk/co/setech/easybook/model/User.java [hunk 2]
```diff
@@ -13,19 +18,16 @@
...
 @Entity
 @Table(name = "_user")
-public class User implements UserDetails {
-    @Id
-    @GeneratedValue
-    private Integer id;
+public class User extends BaseEntity implements UserDetails {
     private String firstName;
     private String lastName;
-    @Column(unique = true)
```
Comment (amoz28) [inline src/main/java/uk/co/setech/easybook/model/User.java:28]: Why did you remove this?, this is to ensure the email field is unique on the db

File: src/main/java/uk/co/setech/easybook/service/VATService.java
Comment (amoz28) [inline src/main/java/uk/co/setech/easybook/service/VATService.java:1]: I thought we wanted to desist from this pattern and not allow Mr Boma's voice guide us

File: src/main/java/uk/co/setech/easybook/service/impl/AuthenticationService.java [hunk 3]
```diff
@@ -87,12 +89,12 @@ private String handleOtp(User user) {
...
 
     public AuthenticationResponse authentication(AuthenticationRequest request) {
-        authenticationManager.authenticate(
+        Authentication authenticate = authenticationManager.authenticate(
                 new UsernamePasswordAuthenticationToken(
                         request.getEmail(),
                         request.getPassword())
         );
-
+        SecurityContextHolder.getContext().setAuthentication(authenticate);
```
Comment (amoz28) [inline src/main/java/uk/co/setech/easybook/service/impl/AuthenticationService.java:107]: What is this supposed to do ?

File: src/main/java/uk/co/setech/easybook/service/impl/AuthenticationService.java [hunk 4]
```diff
@@ -110,7 +112,7 @@ public AuthenticationResponse authentication(AuthenticationRequest request) {
                 .mapToDouble(InvoiceDto::getTotal)
                 .sum();
 
-//        var recentInvoice = invoiceService.getAllInvoicesWithSize(0,5); TODO
+        var recentInvoice = invoiceService.getAllInvoicesWithSize(0,10, "INVOICE");
```
Comment (amoz28) [inline src/main/java/uk/co/setech/easybook/service/impl/AuthenticationService.java]: Modify this to be both invoices and Estimates

File: src/main/java/uk/co/setech/easybook/service/impl/VATServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    @Override
+    public void deleteVAT(Long id) {
+        var vat = vatRepository.findById(id).orElseThrow(VAT_DOES_NOT_EXIST);
+        vatRepository.delete(vat);
+    }
+
+    private VAT toVAT(VATDto dto, VAT vat) {
+        BeanUtils.copyProperties(dto, vat);
```
Comment (amoz28) [inline src/main/java/uk/co/setech/easybook/service/impl/VATServiceImpl.java]: Hope this will also skip null values while updating

File: src/main/java/uk/co/setech/easybook/service/impl/AuthenticationService.java (hunk 4 above)
Comment (ChibuezePaul) [reply src/main/java/uk/co/setech/easybook/service/impl/AuthenticationService.java]: done

File: src/main/java/uk/co/setech/easybook/service/VATService.java
Comment (ChibuezePaul) [reply src/main/java/uk/co/setech/easybook/service/VATService.java:1]: done

File: src/main/java/uk/co/setech/easybook/service/impl/AuthenticationService.java (hunk 3 above)
Comment (ChibuezePaul) [reply src/main/java/uk/co/setech/easybook/service/impl/AuthenticationService.java:107]: set the user to context after login

File: src/main/java/uk/co/setech/easybook/service/impl/VATServiceImpl.java (hunk 5 above)
Comment (ChibuezePaul) [reply src/main/java/uk/co/setech/easybook/service/impl/VATServiceImpl.java]: amen
